src/bepasty/tests/screenshots.py

```diff
@@ -0,0 +1,203 @@
+        """
+        Setup: Open a mozilla browser, login
+        """
+        self.browser = Firefox()
+        self.browser.get(self.url_base + '/invalid')
+
+    def teardown_class(self):
+        """
+        Tear down: Close the browser
+        """
+        self.browser.quit()
+
+    def wait_present(self, xpath, timeout=10):
+        cond = expected_conditions.presence_of_element_located((By.XPATH, xpath))
+        WebDriverWait(self.browser, timeout).until(cond)
+
+    def screen_shot(self, name, w, h):
+        if not os.path.isdir(self.screenshot_dir):
+            os.mkdir(self.screenshot_dir)
+        self.browser.save_screenshot(
+            '{}/{}-{}x{}.png'.format(self.screenshot_dir, name, w, h)
+        )
+
+    def screen_shots(self, name):
+        for w, h in self.screen_sizes:
+            self.browser.set_window_size(w, h)
+            time.sleep(.1)
+            self.screen_shot(name, w, h)
+
+    def scroll_to_bottom(self):
+        self.set_smallest_window_size()
+        self.browser.execute_script('window.scrollTo(0, document.body.scrollHeight);')
+
+    def set_smallest_window_size(self):
+        # change smallest screen size
+        w, h = self.screen_sizes[0]
+        self.browser.set_window_size(w, h)
+        time.sleep(.1)
+
+    def set_largest_window_size(self):
+        # change largest screen size
+        w, h = self.screen_sizes[-1]
+        self.browser.set_window_size(w, h)
+        time.sleep(.1)
+
+    def toggle_hamburger(self):
+        self.set_smallest_window_size()
+
+        # toggle hamburger menu
+        try:
+            self.browser.find_element_by_xpath('//button[@class="navbar-toggler"]').click()
+        except ElementNotInteractableException:
+            pass
+        time.sleep(.5)
+
+        self.set_largest_window_size()
+
+    def top_screen_shots(self, name):
+        self.screen_shots('{}1'.format(name))
+
+        # open hamburger
+        self.toggle_hamburger()
+
+        self.screen_shots('{}2'.format(name))
+
+        # close hamburger
+        self.toggle_hamburger()
+
+    def error_404(self):
+        # NOTE: 404 error
+        self.screen_shots("01-error404")
+
+    def login(self):
+        self.browser.get(self.url_base)
+
+        # NOTE: login screen, 1 - close hamburger, 2 - open hamburger
+        self.top_screen_shots("02-top")
+
+        token = self.browser.find_element_by_name("token")
+        password = "foo"
+        # login
+        token.send_keys(password)
+        token.submit()
+        self.wait_present("//input[@value='Logout']")
+
+        # NOTE: upload screen, 1 - close hamburger, 2 - open hamburger
+        self.top_screen_shots("03-upload")
+
+        try:
+            self.browser.find_element_by_xpath("//input[@value='Logout']")
+        except NoSuchElementException:
+            raise ValueError("Can't login!!! Create a user 'foo' with the permissions"
+                             "'read' and 'create' in your PERMISSIONS in the config")
```

Comment:
not "user", it is a "secret".

So it becomes:

Can't login! Please edit your config, go to PERMISSIONS setting and add a new secret 'screenshots' with 'read' and 'create' permissions.